File: CodingSamples/Java/Runtime/ReflectionTest1/Shop.java
import java.util.*;

/*
A record is a refrence type which on intantiation produces a value-like
immutable object (whose state cannot be modified). Java compiler translates
a record type to a class containing private final instance fields corresponding
to the paramenters specified in the definintion of the record along with
(1) a parameterized constructor to initialize those fields
(2) methods which return values of those fields
(3) overrides for toString, hashCode and equals methods
*/
record Item(String name, String brand) {}

record Customer(String id, double purchase, int rating) implements Comparable<Customer>{
	
	public int compareTo(Customer that) {
		return this.id.compareTo(that.id);
	}
}

class Shop {

	public Item[] getItems() {
		return new Item[] {
			new Item("cpu", "intel"),
			new Item("ssd", "samsung"),
			new Item("mouse", "logitech"),
			new Item("ddr", "samsung"),
			new Item("cpu", "amd"),
			new Item("motherboard", "intel"),
			new Item("ssd", "seagate"),
			new Item("monitor", "samsung"),
			new Item("keyboard", "logitech")
		};
	}

	public Collection<Customer> getCustomers() {
		var customers = new ArrayList<Customer>();
		customers.add(new Customer("Jayesh", 48000, 3));
		customers.add(new Customer("Harshada", 63000, 4));
		customers.add(new Customer("Aishwarya", 32000, 1));
		customers.add(new Customer("Rishikesh", 85000, 5));
		customers.add(new Customer("Snehal", 54000, 2));
		customers.add(new Customer("Neeta", 72000, 4));
		customers.add(new Customer("Karishma", 23000, 2));
		customers.add(new Customer("Vishal", 39000, 4));
		customers.add(new Customer("Chetna", 92000, 5));
		customers.add(new Customer("Pankaj", 42000, 3));
		return customers;
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest1/DemoApp/Shop.cs
//A record type produces instances commonly used for passing of
//data between methods. The C# compiler translates a record type 
//by default into a class containing read-only properties corresponding
//to its positional parameters along with a parameterized constructor
//for initializing those properties and overrides for ToString, 
//GetHashCode and Equals method. 

record Item(string Name, string Brand, decimal Price=5000, int Stock=100); //an immutable reference type record

//a value type record is mutable unless defined with read-only modifier
readonly record struct Customer(string Id, decimal Purchase, int Rating); 

class Shop
{
	public Item[] GetItems() 
    {
		return new Item[] 
        {
			new Item("cpu", "intel"),
			new Item("ssd", "samsung"),
			new Item("mouse", "logitech"),
			new Item("ddr", "samsung"),
			new Item("cpu", "amd"),
			new Item("motherboard", "intel"),
			new Item("ssd", "seagate"),
			new Item("monitor", "samsung"),
			new Item("keyboard", "logitech")
		};
	}

	public ICollection<Customer> GetCustomers() 
    {
		var customers = new List<Customer>();
		customers.Add(new Customer("Jayesh", 48000, 3));
		customers.Add(new Customer("Harshada", 63000, 4));
		customers.Add(new Customer("Aishwarya", 32000, 1));
		customers.Add(new Customer("Rishikesh", 85000, 5));
		customers.Add(new Customer("Snehal", 54000, 2));
		customers.Add(new Customer("Neeta", 72000, 4));
		customers.Add(new Customer("Karishma", 23000, 2));
		customers.Add(new Customer("Vishal", 39000, 4));
		customers.Add(new Customer("Chetna", 92000, 5));
		customers.Add(new Customer("Pankaj", 42000, 3));
		return customers;
	}

}
